### PR TITLE
feat: django32 support.

### PR DESCRIPTION
**Pull request by awais786**
- Django3.2 support.
- Implemented upgrade packages to sync fresh pypi packages.
- Added tox to run multiple django envs.
- Added github actions to run tests.
- Dropped the usage of deprecated `phantomjs` and replaced it with `ChromeHeadless`
- Added a step in CI to update the `gettext` version and made changes to relevant make target to resolve errors with failing tests.
- Dropped Py27 and Py35 support from CI as they have reached end of life

ISSUE: [BOM-2803](https://openedx.atlassian.net/browse/BOM-2803)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Unit tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (aht007) [inline .github/workflows/ci.yml]: We run CI on PRs pointing to any branch not only master and it is something we are following in all repos across edx. For instance you can have a look at this [CI FILE](https://github.com/edx/edx-submissions/blob/master/.github/workflows/ci.yml)

File: Makefile [hunk 2]
```diff
@@ -50,7 +50,8 @@ release:
 	rm README.rst
 
 extract:
-	find ubcpi/ -iname "*.py" ! -path "ubcpi/test/*" | xargs xgettext --from-code=UTF-8 --default-domain=py --language=Python --force-po --output-dir=build --omit-header
+	ls -la
```
Comment (UsamaSadiq) [inline Makefile]: This instruction was only for testing so this  can  be removed  now.

File: Makefile [hunk 3]
```diff
@@ -66,3 +67,18 @@ pull_translations:
...
+	pip install -q -r requirements/pip-tools.txt
+
+upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
+upgrade: piptools ## Update the requirements/*.txt files with the latest packages satisfying requirements/*.in
+	# Make sure to compile files after any other files they include!
+	pip-compile --upgrade --verbose --rebuild -o requirements/pip-tools.txt requirements/pip-tools.in
+	pip-compile --upgrade --verbose --rebuild -o requirements/base.txt requirements/base.in
+	pip-compile --upgrade --verbose --rebuild -o requirements/test.txt requirements/test.in
+	pip-compile --upgrade --verbose --rebuild -o requirements/dev.txt requirements/dev.in
+	# Let tox control the Django and DRF versions for tests
```
Comment (UsamaSadiq) [inline Makefile]:
```suggestion
	# Let tox control the Django version for tests
```

File: karma.conf.js [hunk 4]
```diff
@@ -98,7 +99,7 @@ module.exports = function(config) {
 
     // start these browsers
     // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
-    browsers: ['PhantomJS'],
+    browsers: ['ChromeHeadless'],
```
Comment (UsamaSadiq) [inline karma.conf.js:102]: Mention the reason why we had to change the browser.

File: requirements/constraints.txt [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+# This file doesn't install any packages. It specifies version constraints
+# that will be applied if a package is needed.
+#
+# When pinning something here, please provide an explanation of why.  Ideally,
+# link to other information that will help people in the future to remove the
+# pin when possible.  Writing an issue against the offending project and
+# linking to it here is good.
+
+
+django<4.0
```
Comment (UsamaSadiq) [inline requirements/constraints.txt]: Constraint should be `django<3.3` instead.
```suggestion
django<3.3
```

File: requirements/test.in [hunk 6]
```diff
@@ -0,0 +1,14 @@
+-c constraints.txt
+-r base.in
```
Comment (UsamaSadiq) [inline requirements/test.in]:
```suggestion
-r base.txt
```

File: requirements/dev.in [hunk 7]
```diff
@@ -0,0 +1,16 @@
+-c constraints.txt
+-r test.in
```
Comment (UsamaSadiq) [inline requirements/dev.in]:
```suggestion
-r test.txt
```

File: karma.conf.js (hunk 4 above)
Comment (aht007) [reply karma.conf.js:102]: Mentioned in the PR description....

**Pull request closed without merging**